orthoinference/src/main/java/org/reactome/orthoinference/PathwayDiagramGenerator.java

```diff
@@ -0,0 +1,71 @@
+package org.reactome.orthoinference;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.pathwaylayout.PredictedPathwayDiagramGeneratorFromDB;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.Collection;
+
+public class PathwayDiagramGenerator {
+
+    private static final Logger logger = LogManager.getLogger();
+
+    private static MySQLAdaptor dba;
+    private static GKInstance speciesInst;
+    private static long personId;
+    private static long referenceSpeciesId;
+
+    /**
+     * This method will go through all reference species PathwayDiagrams and finds the orthologous Pathway instances that are for the current target species and
+     * generates the orthologous PathwayDiagram via the PredictedPathwayDiagramGeneratorFromDB method in CuratorTool.
+     * @throws Exception
+     */
+    public static void generateOrthologousPathwayDiagrams() throws Exception {
+
+        logger.info("Generating diagrams for inferred " + speciesInst.getDisplayName() + "pathways");
+        // Create PredictedPathwayDiagramGeneratorFromDB object and set db adaptor and author ID.
+        PredictedPathwayDiagramGeneratorFromDB diagramGenerator = new PredictedPathwayDiagramGeneratorFromDB();
+        diagramGenerator.setMySQLAdaptor(dba);
+        diagramGenerator.setDefaultPersonId(personId);
+
+        GKInstance referenceSpeciesInst = dba.fetchInstance(referenceSpeciesId);
+
+        // Iterate through each PathwayDiagram instance looking for those associated with the reference species.
+        for (GKInstance diagramInst: (Collection<GKInstance>) dba.fetchInstancesByClass(ReactomeJavaConstants.PathwayDiagram)) {
```

Comment:
There is a method, `generateDiagramsForOtherSpecies` inside `PredictedPathwayDiagramGeneratorFromDB` which generates diagrams for a given species. It looks like the code in `EventsInferrer` is passing a species to _this_ class, `PathwayDiagramGenerator`.  From my point of view, it looks like there is some duplication of functionality: Your code and the code in CuratorTool will generate diagrams on a per-species basis.

Is there a reason that you couldn't use `PredictedPathwayDiagramGeneratorFromDB.generateDiagramsForOtherSpecies`? If that method would work in this situation, it would simplify this PR significantly. If that method _cannot_ be used, could you add a brief note or comment explaining why?

Reply:
Yea so that method is named a bit misleadingly. When you supply an id for an 'other' species, it will try to use that species as the reference. Since we always want Human as the reference, and since we want to only create diagrams for inferred Pathways in the target species of interest, we need to go a bit deeper. The generatePredictedDiagram method is the actual method that generates the orthologous diagram, but you need to first have the reference diagram, pathway and orthologous pathway instances.